From 59b431a1b545c2ede92a983ae434166390bc218b Mon Sep 17 00:00:00 2001
From: Eric Chanudet <chanudete@ainfosec.com>
Date: Mon, 1 Jul 2019 15:47:50 -0400
Subject: [PATCH] [efi] Fix sanity check in dbg_efi_protocols.

A null handle would have efi_handle_name() return NULL all the time,
might as well clarify the debug message.

Signed-off-by: Eric Chanudet <chanudete@ainfosec.com>

Modified for ipxe revision 18dc73d27e:
Signed-off-by: Christopher Clark <christopher.clark6@baesystems.com>
diff --git a/src/interface/efi/efi_debug.c b/src/interface/efi/efi_debug.c
index de9b1af5..d0dc6051 100644
--- a/src/interface/efi/efi_debug.c
+++ b/src/interface/efi/efi_debug.c
@@ -280,8 +280,8 @@ void dbg_efi_openers ( EFI_HANDLE handle, EFI_GUID *protocol ) {
 	/* Sanity check */
 	if ( ( ! handle ) || ( ! protocol ) ) {
 		printf ( "HANDLE %s could not retrieve openers for %s\n",
-			 efi_handle_name ( handle ),
-			 efi_guid_ntoa ( protocol ) );
+			 handle ? efi_handle_name ( handle ) : "(null)",
+			 protocol ? efi_guid_ntoa ( protocol ) : "(null)" );
 		return;
 	}
 
